pkg/cortex/cortex.go

```diff
@@ -92,6 +92,9 @@ type Config struct {
 	PrintConfig bool                   `yaml:"-"`
 	HTTPPrefix  string                 `yaml:"http_prefix"`
 
+	ExternalQueryable		prom_storage.Queryable		`yaml:"-"`
+	ExternalPusher 			ruler.Pusher				`yaml:"-"`
```

Comment:
Could you also add links to where this is used in your POC project?

Also add more details in the PR description about how we can initialize a Cortex instance with External implementation.